pulp_rpm/tests/functional/api/test_pulpimport.py

```diff
@@ -0,0 +1,214 @@
+    gen_rpm_client,
+    gen_rpm_remote,
+)
+
+from pulpcore.tests.functional.utils import monitor_task_group
+
+from pulpcore.client.pulpcore import (
+    ApiClient as CoreApiClient,
+    ExportersPulpApi,
+    ExportersCoreExportsApi,
+    ImportersPulpApi,
+    ImportersCoreImportsApi,
+)
+
+from pulpcore.client.pulp_rpm import (
+    RepositoriesRpmApi,
+    RpmRepositorySyncURL,
+    RemotesRpmApi,
+)
+
+NUM_REPOS = 2
+
+
+class PulpImportTestCase(PulpTestCase):
+    """
+    Base functionality for PulpImporter and PulpImport test classes.
+    """
+
+    @classmethod
+    def _setup_repositories(cls):
+        """Create and sync a number of repositories to be exported."""
+        # create and remember a set of repo
+        import_repos = []
+        export_repos = []
+        remotes = []
+        for r in range(NUM_REPOS):
+            import_repo = cls.repo_api.create(gen_repo())
+            export_repo = cls.repo_api.create(gen_repo())
+            body = gen_rpm_remote()
+            remote = cls.remote_api.create(body)
+            repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+            sync_response = cls.repo_api.sync(
+                export_repo.pulp_href, repository_sync_data
+            )
+            monitor_task(sync_response.task)
+            # remember it
+            export_repos.append(export_repo)
+            import_repos.append(import_repo)
+            remotes.append(remote)
+        return import_repos, export_repos, remotes
+
+    @classmethod
+    def _create_exporter(cls, cleanup=True):
+        body = {
+            "name": uuid4(),
+            "repositories": [r.pulp_href for r in cls.export_repos],
+            "path": "/tmp/{}".format(uuid4()),
+        }
+        exporter = cls.exporter_api.create(body)
+        return exporter
+
+    @classmethod
+    def _create_export(cls):
+        export_response = cls.exports_api.create(cls.exporter.pulp_href, {})
+        monitor_task(export_response.task)
+        task = cls.client.get(export_response.task)
+        resources = task["created_resources"]
+        export_href = resources[0]
+        export = cls.exports_api.read(export_href)
+        return export
+
+    @classmethod
+    def _create_chunked_export(cls):
+        export_response = cls.exports_api.create(
+            cls.exporter.pulp_href, {"chunk_size": "5KB"}
+        )
+        monitor_task(export_response.task)
+        task = cls.client.get(export_response.task)
+        resources = task["created_resources"]
+        export_href = resources[0]
+        export = cls.exports_api.read(export_href)
+        return export
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.core_client = CoreApiClient(configuration=cls.cfg.get_bindings_config())
+        cls.file_client = gen_rpm_client()
```

Comment:
s/file_client/rpm_client/

Reply:
ah, nice catch - failure of manual search-and-replace :)